### PR TITLE
RavenDB-13239 Admin Logs - Add 'copy log message' button

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts [hunk 1]
```diff
@@ -210,6 +213,12 @@ class adminLogs extends viewModelBase {
             
             this.duringManualScrollEvent = false;
         });
+      
+        $( ".list-view" ).on( "click", "a", function(event) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: remove space after (

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: 2x

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts [hunk 2]
```diff
@@ -210,6 +213,12 @@ class adminLogs extends viewModelBase {
...
+            event.preventDefault();
+            event.stopImmediatePropagation();
+            copyToClipboard.copy($(this).parent().children().last().text(), "Log message has been copied to clipboard");
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: I think we should use better selector  - this is really fragile

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts [hunk 3]
```diff
@@ -194,9 +194,12 @@ class adminLogs extends viewModelBase {
     
     // noinspection JSMethodCanBeStatic
     itemHtmlProvider(item: string) {
+        const errorClass = this.hasError(item) ? "class='bg-danger'" : "";
+
         return $("<pre class='item'></pre>")
-            .toggleClass("bg-danger", this.hasError(item))
-            .text(item);
+            .addClass("flex-horizontal")
+            .prepend(`<span ${errorClass}>${item}</span>`)
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: do we need to escaping something here?

**Review by ml054 — CHANGES_REQUESTED** [review]

**Comment by ml054** [comment]
+ as we discuss:
- bring back red colors with shades when errors
- remove double scrolls when error items

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: done - and also:
now using `$(".list-view").on("click", ".copy-item-button", function(event) { ....`
(changed class name in other places where needed)

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: now using: `$(this).next().text()`

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: no - I don't think we need to

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 2 above)
Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: can we use $("span", this).text() ?

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: done

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: The dom structure per item is:

```
<div>
    <pre>
        <a>the copy button</a>
        <span>the text to copy</span>
    </pre>
</div>
```

`$(this)` is the `<a>` element that was clicked on - it has no text on it.
so I go to `$(this).next()` - which is the `<span>` that has the text.

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: oh ok, so `$(this).next().text()` should be good

File: src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts (hunk 1 above)
Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/manage/adminLogs.ts]: yes - this is nice and less fragile

**Review by ml054 — APPROVED** [review]

**Pull request merged**